osu.Game.Tests/Rulesets/Scoring/ScoreProcessorTest.cs

```diff
@@ -53,5 +53,193 @@ public void TestSingleOsuHit(ScoringMode scoringMode, HitResult hitResult, int e
 
             Assert.IsTrue(Precision.AlmostEquals(expectedScore, scoreProcessor.TotalScore.Value));
         }
+
+        [TestCase(ScoringMode.Standardised, HitResult.Miss, HitResult.Great, 0)]
+        [TestCase(ScoringMode.Standardised, HitResult.Meh, HitResult.Great, 387_500)]
+        [TestCase(ScoringMode.Standardised, HitResult.Ok, HitResult.Great, 425_000)]
+        [TestCase(ScoringMode.Standardised, HitResult.Good, HitResult.Perfect, 3_350_000 / 7.0)]
+        [TestCase(ScoringMode.Standardised, HitResult.Great, HitResult.Great, 575_000)]
+        [TestCase(ScoringMode.Standardised, HitResult.Perfect, HitResult.Perfect, 575_000)]
+        [TestCase(ScoringMode.Standardised, HitResult.SmallTickMiss, HitResult.SmallTickHit, 0)]
+        [TestCase(ScoringMode.Standardised, HitResult.SmallTickHit, HitResult.SmallTickHit, 0)] // TODO: this should probably be 225_000 from accuracy portion
```

Comment:
because you don't seem to have looked into this in any capacity, I will explain why this is passing (probably wrongly). this is due to the following conditional in `ScoreProcessor`.

https://github.com/ppy/osu/blob/c803be2dd260ce5d1963a53c08d9f48bbc9c8d35/osu.Game/Rulesets/Scoring/ScoreProcessor.cs#L271-L275

because the maximum combo is 0 due to all judgements not affecting combo, the scoring mode is forced to `Classic`, which means that the score will be always 0 if there are no bonuses. mind you, that is probably wrong, but I don't think it's acceptable to submit this test without looking into why this is not working correctly at all.

Reply:
I know there was a reason for that at some point... But I think the relevant codepath is properly handled now:
https://github.com/ppy/osu/blob/f206ea96ce6ccbce87af0e360b0f2630ceac0635/osu.Game/Rulesets/Scoring/ScoreProcessor.cs#L196-L202

Could the forcing of `Classic` just be removed?